xds/src/main/java/io/grpc/xds/XdsComms2.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Supplier;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment;
+import io.envoyproxy.envoy.api.v2.DiscoveryRequest;
+import io.envoyproxy.envoy.api.v2.DiscoveryResponse;
+import io.envoyproxy.envoy.api.v2.core.Node;
+import io.envoyproxy.envoy.service.discovery.v2.AggregatedDiscoveryServiceGrpc;
+import io.grpc.ChannelLogger.ChannelLogLevel;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext.ScheduledHandle;
+import io.grpc.internal.BackoffPolicy;
+import io.grpc.stub.StreamObserver;
+import io.grpc.xds.LookasideChannelLb.AdsStreamCallback2;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.CheckForNull;
+
+/**
+ * ADS client implementation.
+ */
+// TODO(zdapeng): This is a temporary and easy refactor of XdsComms, will be replaced by XdsClient.
+// Tests are deferred in XdsClientTest, otherwise it's just a refactor of XdsCommsTest.
+final class XdsComms2 {
```

Comment:
Please clean up this class. 

- Put fields and constructors together.
- Remove/reformat unnecessary comments, let the code explain itself.
- Do not overly nest classes, make them as flat as possible. Nests make indentation bad and hard to read.

Reply:
As for comments to `XdsComms2`, `XdsComms2` is only a temporary refactor of `XdsComms`, it will be re-implemented with the new XdsClient design. Cleaning up existing `XdsComms` code is not part of this PR.

Reply:
Yes I mean `XdsComms2`. Ok, make things cleaner when moving to XdsClient.